### PR TITLE
MGMT-9460: Always use the image service in inventory unit tests

### PR DESCRIPTION
**Pull request by carbonin**
Previously these tests defaulted to testing the old s3 code which is no
longer used in any production environment.

Changing these tests will make removing the old s3 code much easier

## List all the issues related to this PR

https://issues.redhat.com/browse/MGMT-9460

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [x] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @nmagnezi 
/cc @danielerez 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

### CODE REVIEW

**Comment by carbonin** [comment]
Just a note for reviewers here.
Reviewing with ignoring whitespace will be helpful because I removed a context so I had to change the indentation level of a ton of code.

File: internal/bminventory/inventory_test.go [hunk 1]
```diff
@@ -14039,14 +13962,6 @@ var _ = Describe("GetInfraEnvDownloadURL", func() {
 			Expect(infraEnv.ExpiresAt.Equal(payload.ExpiresAt)).To(BeTrue())
 		})
 	})
-
-	It("when not using the image service it returns 400", func() {
```
Comment (nmagnezi) [inline internal/bminventory/inventory_test.go:14043]: That's no longer an option? do we enforce it?

Comment (carbonin) [reply internal/bminventory/inventory_test.go:14043]: Yes to both: https://github.com/openshift/assisted-service/blob/9f463c8ac54821c57089dd68a86a27fce43b4e5b/cmd/main.go#L204-L206

**Pull request merged**